---
title: "Population_Genetics"
author: "Quinn W"
date: "6/1/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(poppr)
library(vcfR)
library(tidyverse)
library(ape)
library(RColorBrewer)
library(igraph)
```

```{r}

```


```{r}
#read in as vcfR
lox_pooled_98_file <- "./Data/loxodes_pooled/all_filtered_98.vcf"
lox_pooled_98_vcfR <- read.vcfR(lox_pooled_98_file)

lox_pooled_98_wga <- "./Data/loxodes_genome/all_filtered_wga_98.vcf"
lox_pooled_98_wga_vcfR <- read.vcfR(lox_pooled_98_wga)

halt_90 <- read.vcfR("./Data/halteria/all_filtered_halt_90.vcf")
chilo_90 <- read.vcfR("./Data/chilo/all_filtered_chilo_90.vcf")


#convert to genlight object - only considers bi-allelic SNPs

#transcriptome 
lox_pooled_genlight <- vcfR2genlight(lox_pooled_98_vcfR)

pop(lox_pooled_genlight) <- substr(indNames(lox_pooled_genlight), 1, 6)

#genome
lox_pooled_wga_genlight <- vcfR2genlight(lox_pooled_98_wga_vcfR)

pop(lox_pooled_wga_genlight) <- substr(indNames(lox_pooled_wga_genlight), 1, 6)

#halt
halt_genlight <- vcfR2genlight(halt_90)
pop(halt_genlight) <- substr(indNames(halt_genlight), 1, 7)

#chilodonella
chilo_genlight <- vcfR2genlight(chilo_90)
pop(chilo_genlight) <- substr(indNames(chilo_genlight), 1, 7)




```

Here, we lost 31 loci that weren't bi-allelic, because genlight objects only support bi-allelic loci.

Genlight objects can support individuals of different ploidies, but within a single individual the ploidy must be the same for all loci. 

The assumptions for the bitwise.dist() function are that SNPs are bi-allelic, samples are haploid/diploid, and all samples have the same ploidy. 

```{r}
#set ploidy
ploidy(lox_pooled_genlight) <- 2
ploidy(lox_pooled_wga_genlight) <- 2
ploidy(halt_genlight) <- 2
ploidy(chilo_genlight) <- 2


#form distance matrix with the bitwise.dist function from poppr
lox_dist <- bitwise.dist(lox_pooled_genlight)
lox_dist_wga <- bitwise.dist(lox_pooled_wga_genlight)
halt_dist <- bitwise.dist(halt_genlight)
chilo_dist <- bitwise.dist(chilo_genlight)



```

Now, we'll create a minimum spanning network to visualize the population structure. The MSN groups multilocus genotypes by the genetic distances between them.The multilocus genotypes are the nodes, and the lengths of the branches represent the genetic distance.

```{r}
#transcriptome
lox_msn <- poppr.msn(lox_pooled_genlight, lox_dist, showplot = FALSE, include.ties = T)

#genome
lox_msn_wga <- poppr.msn(lox_pooled_wga_genlight, lox_dist_wga, showplot = FALSE, include.ties = T)
#halteria
halt_msn <- poppr.msn(halt_genlight, halt_dist, showplot = FALSE, include.ties = T)
#chilodonella
chilo_msn <- poppr.msn(chilo_genlight, chilo_dist, showplot = FALSE, include.ties = T)


lox_msn$populations <- substr(lox_msn$populations, 1, 6)

halt_msn$populations <- substr(halt_msn$populations, 1, 7)

chilo_msn$populations <- substr(chilo_msn$populations, 1, 7)

#transcriptome
node_size <- rep(2, times = nInd(lox_pooled_genlight))
names(node_size) <- popNames(lox_pooled_genlight)
lox_pooled_genlight$ind.names <- substr(lox_pooled_genlight$ind.names, 1, 6)

vertex_attr(lox_msn$graph)$size <- node_size

set.seed(9)
plot_poppr_msn(lox_pooled_genlight, lox_msn, palette = brewer.pal(n = nPop(lox_pooled_genlight), name = "Set3"), vertex.label.dist = 1, gadj = 70, pop.leg = FALSE, size.leg = FALSE) 

#genome 
node_size <- rep(2, times = nInd(lox_pooled_wga_genlight))

names(node_size) <- popNames(lox_pooled_wga_genlight)

lox_pooled_wga_genlight$ind.names <- substr(lox_pooled_wga_genlight$ind.names, 1, 6)

vertex_attr(lox_msn_wga$graph)$size <- node_size

set.seed(9)

plot_poppr_msn(lox_pooled_wga_genlight, lox_msn_wga, palette = brewer.pal(n = nPop(lox_pooled_wga_genlight), name = "Set3"), vertex.label.dist = 1, gadj = 450, pop.leg = FALSE, size.leg = FALSE) 

#halteria
node_size <- rep(2, times = nInd(halt_genlight))
names(node_size) <- popNames(halt_genlight)
halt_genlight$ind.names <- substr(halt_genlight$ind.names, 1, 7)

vertex_attr(halt_msn$graph)$size <- node_size

set.seed(9)

plot_poppr_msn(halt_genlight, halt_msn, palette = brewer.pal(n = nPop(halt_genlight), name = "Set3"), vertex.label.dist = 1, gadj = 70, pop.leg = FALSE, size.leg = FALSE) 


#chilodonella
node_size <- rep(2, times = nInd(chilo_genlight))
names(node_size) <- popNames(chilo_genlight)
chilo_genlight$ind.names <- substr(chilo_genlight$ind.names, 1, 7)

vertex_attr(chilo_msn$graph)$size <- node_size

set.seed(9)

plot_poppr_msn(chilo_genlight, chilo_msn, palette = brewer.pal(n = nPop(halt_genlight), name = "Set3"), vertex.label.dist = 1, gadj = 70, pop.leg = FALSE, size.leg = FALSE) 


```

Now we can perform principal components analysis (PCA). This involves a rotation the axes to maximize variance between the samples. The SNP data is transformed into principal components, which are uncorrelated variables that are linear combinations of the initial variables. Maximum information is put into the first components. These principal components are the axes that represent the maximal amount of variance.

```{r}

#remove the NA values

#transcriptome
remove <- is.na(glMean(lox_pooled_genlight, alleleAsUnit = FALSE))
which(remove)
lox_pooled_genlight_new <- lox_pooled_genlight[, !remove]

#genome
removewga <- is.na(glMean(lox_pooled_wga_genlight, alleleAsUnit = FALSE))
which(removewga)
lox_pooled_wga_genlight_new <- lox_pooled_wga_genlight[, !removewga]

#halteria
remove <- is.na(glMean(halt_genlight, alleleAsUnit = FALSE))
which(remove)
halt_genlight_new <- halt_genlight[, !remove]

#chilodonella
remove <- is.na(glMean(chilo_genlight, alleleAsUnit = FALSE))
which(remove)
chilo_genlight_new <- chilo_genlight[, !remove]



#perform principal component analysis
#transcriptome
lox_pca <- glPca(lox_pooled_genlight_new, nf = 3)
#plot eigenvalues to determine how many PCAs to retain
barplot(100*lox_pca$eig/sum(lox_pca$eig), main = "PCA Eigenvalues")
title(ylab = "Percent of variance explained", line = 2)
title(xlab = "Eigenvalues", line = 1)
#include the principal components in a data frame
lox_pca_scores <- as.data.frame(lox_pca$scores)
lox_pca_scores$pop <- pop(lox_pooled_genlight)

my_palette <- c("#4287f5", "#42f5e3", "#4ebf69", "#cf9c2d", "#f25d1d", "#b81111", "#1f2194", "#922ce6", "#b52a5f", "#416b69")
lox_plot <- ggplot(lox_pca_scores, aes(x=PC1, y=PC2, colour = pop)) + 
  geom_point(size=2) +
  geom_hline(yintercept=0) + 
  geom_vline(xintercept=0) + 
  theme_minimal() + 
  scale_color_manual(values = my_palette) + 
  labs(title = "PCA Analysis for Loxodes (Transcriptome)") + theme(plot.title = element_text(face = "bold", hjust = .5),
                                                                   legend.title = element_blank())

lox_plot

#genome
lox_pca_wga <- glPca(lox_pooled_wga_genlight_new, nf = 3)
#plot eigenvalues to determine how many PCAs to retain
barplot(100*lox_pca_wga$eig/sum(lox_pca_wga$eig), main = "PCA Eigenvalues")
title(ylab = "Percent of variance explained", line = 2)
title(xlab = "Eigenvalues", line = 1)
#include the principal components in a data frame
lox_pca_scores_wga <- as.data.frame(lox_pca_wga$scores)
lox_pca_scores_wga$pop <- pop(lox_pooled_wga_genlight_new)


lox_plot_wga <- ggplot(lox_pca_scores_wga, aes(x=PC1, y=PC2, colour = pop)) + 
  geom_point(size=2) +
  geom_hline(yintercept=0) + 
  geom_vline(xintercept=0) + 
  theme_minimal() + 
  scale_color_manual(values = my_palette) + 
  labs(title = "PCA Analysis for Loxodes (Genome)") + theme(plot.title = element_text(face = "bold", hjust = .5),
                                                                   legend.title = element_blank())

lox_plot_wga

#halteria
halt_pca <- glPca(halt_genlight_new, nf = 3)
#plot eigenvalues to determine how many PCAs to retain
barplot(100*halt_pca$eig/sum(halt_pca$eig), main = "PCA Eigenvalues")
title(ylab = "Percent of variance explained", line = 2)
title(xlab = "Eigenvalues", line = 1)
#include the principal components in a data frame
halt_pca_scores <- as.data.frame(halt_pca$scores)
halt_pca_scores$pop <- pop(halt_genlight_new)


halt_plot <- ggplot(halt_pca_scores, aes(x=PC1, y=PC2, colour = pop)) + 
  geom_point(size=2) +
  geom_hline(yintercept=0) + 
  geom_vline(xintercept=0) + 
  theme_minimal() + 
  scale_color_manual(values = my_palette) + 
  labs(title = "PCA Analysis for Halteria") + theme(plot.title = element_text(face = "bold", hjust = .5),
                                                                   legend.title = element_blank())

halt_plot

#chilodonella
chilo_pca <- glPca(chilo_genlight_new, nf = 3)
#plot eigenvalues to determine how many PCAs to retain
barplot(100*chilo_pca$eig/sum(chilo_pca$eig), main = "PCA Eigenvalues")
title(ylab = "Percent of variance explained", line = 2)
title(xlab = "Eigenvalues", line = 1)
#include the principal components in a data frame
chilo_pca_scores <- as.data.frame(chilo_pca$scores)
chilo_pca_scores$pop <- pop(chilo_genlight_new)


chilo_plot <- ggplot(chilo_pca_scores, aes(x=PC1, y=PC2, colour = pop)) + 
  geom_point(size=2) +
  geom_hline(yintercept=0) + 
  geom_vline(xintercept=0) + 
  theme_minimal() + 
  scale_color_manual(values = my_palette) + 
  labs(title = "PCA Analysis for Chilodonella") + theme(plot.title = element_text(face = "bold", hjust = .5),
                                                                   legend.title = element_blank())

chilo_plot

#create dendrogram from genind object
aboot(chilo_genlight_new, distance = bitwise.dist, sample = 1000)

aboot(halt_genlight_new, distance = bitwise.dist, sample = 1000)

aboot(lox_pooled_genlight_new, distance = bitwise.dist, sample = 1000)



```

```{r}
library(reshape2)
#locus table analysis

lox_trans_98_gd <- vcfR2genind(lox_pooled_98_vcfR)

lox_trans_98_loc <- locus_table(lox_trans_98_gd, index = "simpson", lev = "allele") %>% as_tibble()

var <- c("1-D")
lox_trans_mdf <- melt(lox_trans_98_loc["1-D"],
              value.name = 'Depth',
              na.rm = TRUE)

lox_trans_mdf %>% 
    ggplot(aes(x = variable, y = Depth)) +
  geom_violin(fill = "#568D55") + 
  theme_minimal() 

```

